syntax = "proto3";

// messages (protocol buffers, aka "protobufs")

message SetCoefsRequest {
	repeated float coefs = 1;
}

message SetCoefsResponse {
	string error = 3;
}

message PredictRequest {
	repeated float X = 1;
}

message PredictResponse {
	float y = 1;
	bool hit = 2;
	string error = 3;
}


// services (RPC functions)

service ModelServer {
	rpc SetCoefs(SetCoefsRequest) returns (SetCoefsResponse);
	rpc Predict(PredictRequest) returns (PredictResponse);
}
